### DISCUSSION TITLE
How to use useNuxtApp() in defineEventHandler?

### DISCUSSION THREAD

**Discussion by besscroft** [nuxt/nuxt#24251] [Questions] [answered]
Hello the community,

I have defined a `server plugins` that I want to be able to use via `const { $sql } = useNuxtApp()`. However, ReferenceError: useNuxtApp is not defined is prompted in defineEventHandler.
Someone can point me to the good direction?

Thanks!

**Comment by TheAlexLichter** [comment, marked as answer]
You don't. Composables cannot to be used in the `server` folder.
Instead, you can define a Nitro plugin, injecting the functionality in `event.context`, or instead importing the desired functions as usual ☺️

**Reply by Bulkman** [reply]
Except useRuntimeConfig right? Because documentation clearly states that we can use it there. We can't use importing either when it contains shared state variables or singletons that accumulates that states, is that correct?

**Reply by TheAlexLichter** [reply]
@Bulkman Yes, that's an exception. `useRuntimeConfig` is "doing the same" but defined on the nitro side independently of Nuxt 😋

You can import singletons or shared state but you have to deal with the side effects accordingly then. So - ideally not, but sometimes you have no "other choice" :D
